### PR TITLE
Studio: Add rotation to welcome messages

### PR DESCRIPTION
**Pull request by katinthehatsite**
Closes: https://github.com/Automattic/dotcom-forge/issues/8981

## Proposed Changes

This PR adds a rotation for example prompts so that every time you access the Assistant tab, you will get a new selection of prompts that come from the backend.

## Testing Instructions

* Apply this diff to your sandbox: D160843-code
* Sandbox the public API
* Pull the changes from this branch
* Start the app with `nvm use && npm install && npm start`
* Create a site
* Navigate to the Assistant tab
* Observe the two welcome messages and three example prompts
* Switch to the `Settings` tab
* Navigate back to the `Assistant` tab
* Observe the new set of three example prompts 

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/hooks/use-rotate-promp-messages.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { useWelcomeMessages } from '../hooks/use-welcome-messages';
+
+export function useRotateWelcomeMessages( siteId: string ) {
+	const { examplePrompts } = useWelcomeMessages();
+
+	//Limit to 3 and shuffle initial prompts
+	const randomizedPrompts = useMemo( () => {
+		const shuffled = [ ...examplePrompts ].sort( () => 0.5 - Math.random() );
+		return shuffled.slice( 0, 3 );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (katinthehatsite) [inline src/hooks/use-rotate-promp-messages.ts:11]: I disabled eslint for this line as it gives warning for siteId not being used but we want to include it to make sure prompts change for each site.

If there is a better solution, happy to apply that!

**Comment by katinthehatsite** [comment]
I am expecting that the rotation mechanism might become more complex in the future so I added it to a separate hook

**Review by sejas — COMMENTED** [review]
Kat, great work!

I tested it and I confirm the welcome messages rotate.
I have a small concern about the prompts rotating when the user changes the tab or even when the focus changes between windows.

What do you think? Should we keep them static once they've loaded and change them only when the user rests the conversation?

https://github.com/user-attachments/assets/b2aed515-01b6-4361-a466-7bb97b3ee09f

**Comment by matt-west** [comment]
> Should we keep them static once they've loaded and change them only when the user rests the conversation?

I think that's sensible. It definitely feels weird to me for the prompts to change when the window is re-focussed.

**Comment by katinthehatsite** [comment]
@sejas we can cache the rotation once the user opens them for the first time on a specific site. I would like to merged this first if possible so that I can also work on https://github.com/Automattic/dotcom-forge/issues/8982 as some of the code is going to overlap. Can we handle this in a separate PR?

**Review by sejas — APPROVED** [review]
Yes!, let's do that. That way we could showcase this feature in user interviews.

- I've created another issue to make the rotation fixed per site: https://github.com/Automattic/dotcom-forge/issues/9102

Great job kat!

**Pull request merged**